packages/transaction-controller/src/utils/swaps.test.ts

```diff
@@ -375,6 +375,60 @@ describe('updateSwapsTransaction', () => {
       swapAndSendRecipient,
     });
   });
+
+  it('should update swap and send approval transaction and publish TransactionController:transactionNewSwapAndSendApproval', async () => {
+    const sourceTokenSymbol = 'ETH';
+    const type = TransactionType.swapAndSendApproval;
+
+    swaps.meta = {
+      sourceTokenSymbol,
+      type,
+    };
+    transactionType = TransactionType.swapAndSendApproval;
+
+    const transactionNewSwapAndSendApprovalEventListener = jest.fn();
+    messenger.subscribe(
+      'TransactionController:transactionNewSwapAndSendApproval',
+      transactionNewSwapAndSendApprovalEventListener,
+    );
+
+    updateSwapsTransaction(transactionMeta, transactionType, swaps, request);
+    expect(
+      transactionNewSwapAndSendApprovalEventListener,
+    ).toHaveBeenCalledTimes(1);
+    expect(transactionNewSwapAndSendApprovalEventListener).toHaveBeenCalledWith(
+      {
+        transactionMeta: {
+          ...transactionMeta,
+          sourceTokenSymbol,
+          type,
+        },
+      },
+    );
+  });
+
+  it('should return the swap and send approval transaction updated with information', async () => {
```

Comment:
[nit] can remove async
```suggestion
  it('should return the swap and send approval transaction updated with information', () => {
```